src/stx/astf/trex_astf_rpc_cmds.cpp

```diff
@@ -332,29 +332,30 @@ trex_rpc_cmd_rc_e
 TrexRpcCmdAstfProfileList::_run(const Json::Value &params, Json::Value &result) {
     vector<string> profile_list = get_astf_object()->get_profile_id_list();
     Json::Value json_profile_list = Json::arrayValue;
+    bool profiles_in_ports = parse_bool(params, "profiles_in_ports", result, false);
 
     for (auto &profile_id : profile_list) {
         json_profile_list.append(profile_id);
     }
 
-    uint8_t port_cnt = get_astf_object()->get_port_count();
-    for (uint8_t port_id = 0; port_id < port_cnt; port_id++) {
-        TrexStatelessPort *port = get_stateless_obj()->get_port_by_id(port_id);
-        profile_list.clear();
+    if (profiles_in_ports) {
```

Comment:
I don't see the code in case that profiles_in_ports==False (backward compatible)

Reply:
@hhaim this change is for trex-java-sdk's backward compatible.
ASTF extends STL feature #936 added the STL profiles in ports by default.
It result of the command changed from [ "_" ] to [ "_", { "\u0000": [ "_" ] }, { "\u0001": [ "_" ] } ].
The Python client doesn't care about the result.
But, trex-java-sdk client treats the result as an array of strings. It causes an exception in the client.
By this change, the result is back to [ "_" ] and the exception will be suppressed.

Reply:
@jsmoon understood